code/modules/projectiles/guns/energy.dm

```diff
@@ -131,6 +134,18 @@
 		update_icon()
 
 /obj/item/gun/energy/attackby(obj/item/C, mob/living/user)
+	if(QUALITY_SAWING in C.tool_qualities)
+		to_chat(user, SPAN_NOTICE("You begin to saw down \the [src]."))
+		if(saw_off == FALSE)
+			to_chat(user, SPAN_NOTICE("Sawing down \the [src] will achieve nothing or may impede operation."))
+			return
+		if (src.item_upgrades.len)
+			if("No" == input(user, "There are attachments present. Would you like to destroy them?") in list("Yes", "No"))
```

Comment:
Pretty sure this can be used to cheese DNA lock trigger and other ones, might want to either add a check for those or disallow sawing off if there's an attachment altogether

Reply:
should be taken care of